Tasks/UseDotNetV2/usedotnet.ts

```diff
@@ -2,52 +2,84 @@
     let versionSpec = tl.getInput('version');
-    if (versionSpec) {
-        console.log(tl.loc("ToolToInstall", packageType, versionSpec));
+    const nugetVersion = tl.getInput('nugetVersion') || '4.4.1';
+
+    // Check if we want install dotnet
+    if (versionSpec || useGlobalJson) {
+        let packageType = tl.getInput('packageType') || "sdk";
         let installationPath = tl.getInput('installationPath');
-        if (!installationPath) {
-            installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
-        }
         let includePreviewVersions: boolean = tl.getBoolInput('includePreviewVersions');
+        let workingDirectory: string | null = tl.getPathInput("workingDirectory", false) || null;
+        let performMultiLevelLookup = tl.getBoolInput("performMultiLevelLookup", false);
+        await installDotNet(installationPath, packageType, versionSpec, useGlobalJson, workingDirectory, includePreviewVersions, performMultiLevelLookup);
+    }
+    // Install NuGet version specified by user or 4.4.1 in case none is specified
+    // Also sets up the proxy configuration settings.    
+    await NuGetInstaller.installNuGet(nugetVersion);
+}
 
-        var versionSpecParts = new VersionParts(versionSpec);
+/**
+ * install dotnet to the installation path.
+ * @param installationPath The installation path. If this is empty it would use {Agent.ToolsDirectory}/dotnet/
+ * @param packageType The installation type for the installation. Only `sdk` and `runtime` are valid options
+ * @param versionSpec The version the user want to install.
+ * @param useGlobalJson A switch so we know if the user have `global.json` files and want use that.
+ * @param workingDirectory This is only relevant if the `useGlobalJson` switch is `true`. It will set the root directory for the search of `global.json`
+ * @param includePreviewVersions Define if the installer also search for preview version
+ * @param performMultiLevelLookup Set the `DOTNET_MULTILEVEL_LOOKUP`environment variable
+ */
+async function installDotNet(
+    installationPath: string,
+    packageType: string,
+    versionSpec: string | null,
+    useGlobalJson: boolean,
+    workingDirectory: string | null,
+    includePreviewVersions: boolean,
+    performMultiLevelLookup: boolean) {
 
-        let versionFetcher = new DotNetCoreVersionFetcher();
+    if (!installationPath && installationPath.length == 0) {
+        installationPath = path.join(tl.getVariable('Agent.ToolsDirectory'), "dotnet");
+    }
+    let versionFetcher = new DotNetCoreVersionFetcher();
+    let dotNetCoreInstaller = new VersionInstaller(packageType, installationPath);
+    if (useGlobalJson) {
+        let globalJsonFetcherInstance = new globalJsonFetcher(workingDirectory);
+        let versionsToInstall: VersionInfo[] = await globalJsonFetcherInstance.Get(packageType);        
+        versionsToInstall = versionsToInstall.filter(d => !dotNetCoreInstaller.isVersionInstalled(d.getVersion()));
```

Comment:
Can a version be repeated twice in two global json files. It is better to iterate over all the versions and filter isVersionInstalled inside the for loop so that such a version is skipped there

Reply:
yes this is a case. 

I think we should do both.

Reply:
What is the benefit of doing it outside of the for loop. I think we can just do it inside the loop. And you can move the logic to getDownloadUrl inside the condition of isVersionIntalled